### PR TITLE
Make EmployeeId accept integers only

### PR DESCRIPTION
**Pull request by gavinkalika**
Create unit test for value object in effort to learn code base

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | Create unit test for value object in effort to learn code base
| Type?         | improvement 
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | (optional) If this PR fixes an [Issue](https://github.com/PrestaShop/PrestaShop/issues), please write "Fixes #[issue number]" here.
| How to test?  | Execute unit test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15010)

### CODE REVIEW

File: src/Core/Domain/Employee/ValueObject/EmployeeId.php [hunk 1]
```diff
@@ -63,7 +65,7 @@ public function getValue()
      */
     private function assertIntegerIsGreaterThanZero($employeeId)
     {
-        if (!is_numeric($employeeId) || 0 > $employeeId) {
```
Comment (gavinkalika) [inline src/Core/Domain/Employee/ValueObject/EmployeeId.php:66]: this method accepts 1.1111 as a valid id.

That is why I removed it.

**Comment by gavinkalika** [comment]
> * category

what is category?

**Comment by MathiasReker** [comment]
Category is `CO`

**Comment by gavinkalika** [comment]
I don’t understand why the integration tests failed. I didn’t touch those

File: tests/Unit/Core/Domain/Employee/ValueObject/EmployeeIdTest.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Unit\Core\Domain\Employee\ValueObject;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Domain\Employee\Exception\InvalidEmployeeIdException;
+use PrestaShop\PrestaShop\Core\Domain\Employee\ValueObject\EmployeeId;
+
+class EmailTest extends TestCase
```
Comment (rokaszygmantas) [inline tests/Unit/Core/Domain/Employee/ValueObject/EmployeeIdTest.php]:
```suggestion
class EmployeeIdTest extends TestCase
```

Comment (gavinkalika) [reply tests/Unit/Core/Domain/Employee/ValueObject/EmployeeIdTest.php]: yikes!!! 💩

**Comment by gavinkalika** [comment]
interesting the build still failed.

Is this an allowed failure? I inspected the Travis CI build

**Comment by matks** [comment]
@gavinkalika sorry about that, this is a random failure that happens
we'll fix it, we created issue https://github.com/PrestaShop/PrestaShop/issues/14497 for that :/

we re-run the tests as it fails randomly, your code is good

**Comment by gavinkalika** [comment]
@matks thanks and good to know. Perhaps I'll take a look into that bug when I get the time.

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
@gavinkalika the PR makes sense indeed 👍 

@sarjon @tomas862 @rokaszygmantas any reason why we used `is_numeric` instead of `is_int` ? When I look at it it seems obvious this is the right way, but maybe I have forgotten one obscure usecase 🤔 ?

Note to myself: I'll have to change all `is_numeric` to `is_int` we used for ID VOs then 😛

**Comment by sarjon** [comment]
It could be related to creating VOs with string `new EmployeeId('1')`, but I might be wrong. Either way, this fix looks good to me. :slightly_smiling_face:

**Comment by matks** [comment]
A big thank you @gavinkalika, it's not everyday that we have someone contributing to the tests 😉

**Comment by matks** [comment]
I created ticket https://github.com/PrestaShop/PrestaShop/issues/15145 to apply this change everywhere

**Pull request merged**